engine/consensus/dkg/messaging_engine.go

```diff
@@ -76,7 +76,12 @@ func (e *MessagingEngine) Done() <-chan struct{} {
 
 // SubmitLocal implements the network Engine interface
 func (e *MessagingEngine) SubmitLocal(event interface{}) {
-	e.Submit(engine.DKGCommittee, e.me.NodeID(), event)
+	e.unit.Launch(func() {
+		err := e.Process(engine.DKGCommittee, e.me.NodeID(), event)
+		if err != nil {
+			engine.LogError(e.log, err)
```

Comment:
three requests for revisions:
* Here, you are calling `unit.Launch` which subsequently calls `e.Process` which then calls `unit.Do`. I find this is an unnecessary duplication of `unit` calls. 
* If you look at the implementation of `Process`, it is straight forward to convince yourself that the only error it can throw is an invalid input https://github.com/onflow/flow-go/blob/1f463cc9c73092c52622a1f7447b721a4563bd60/engine/consensus/dkg/messaging_engine.go#L120 If a local, _trusted_ component feeds an event of invalid type into this engine, that is certainly a fatal implementation bug. 
* formally, `engine.DKGCommittee` is an identifier for a _networking_ channel. `SubmitLocal` receives events from an _internal engine_, which is _not_ a networking channel. 


the following would address all three concerns:
```suggestion
		err := e.process(originID, event)
		if err != nil {
			e.log.Fatal().Err(err).Msg("unexpected internal error processing event")
```